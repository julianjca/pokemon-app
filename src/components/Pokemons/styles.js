import styled from "@emotion/styled";

export const Grid = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 3rem;

  @media (min-width: 720px) {
    grid-template-columns: repeat(3, 1fr);
  }
`;
