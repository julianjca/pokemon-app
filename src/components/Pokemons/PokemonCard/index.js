import React from "react";
import Link from "next/link";

import { Card, Pokeball, PokemonName } from "./styles";

const PokemonCard = ({ name, image }) => {
  return (
    <Link href={`/pokemons/${name}`}>
      <Card>
        <img src={image} alt={name} />
        <PokemonName>{name}</PokemonName>
        <Pokeball src="/pokeball.svg" />
      </Card>
    </Link>
  );
};

export default PokemonCard;
