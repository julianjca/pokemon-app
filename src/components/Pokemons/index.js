import React from "react";

import PokemonCard from "./PokemonCard";
import { Grid } from "./styles";
import Button from "../Button";

const Pokemons = ({ pokemons, fetchMore }) => {
  return (
    <div
      style={{
        padding: "4rem 0",
      }}
    >
      <Grid>
        {pokemons.results.map((pokemon, key) => {
          return <PokemonCard key={key} {...pokemon} />;
        })}
      </Grid>
      <div
        style={{
          width: "100%",
          textAlign: "center",
          marginTop: "2rem",
        }}
      >
        <Button
          onClick={() =>
            fetchMore({
              variables: {
                offset: pokemons.results.length,
              },
            })
          }
        >
          Load more
        </Button>
      </div>
    </div>
  );
};

export default Pokemons;
