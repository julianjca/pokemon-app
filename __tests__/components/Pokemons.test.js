/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { matchers } from "@emotion/jest";

import { render, fireEvent } from "../../test-utils";
import Pokemons from "../../src/components/Pokemons";

expect.extend(matchers);

HTMLCanvasElement.prototype.getContext = () => {
  // return whatever getContext has to return
};

jest.mock("../../src/lib");

const pokemons = {
  results: [
    {
      __typename: "PokemonItem",
      id: 1,
      url: "https://pokeapi.co/api/v2/pokemon/1/",
      name: "bulbasaur",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
    },
    {
      __typename: "PokemonItem",
      id: 2,
      url: "https://pokeapi.co/api/v2/pokemon/2/",
      name: "ivysaur",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png",
    },
    {
      __typename: "PokemonItem",
      id: 3,
      url: "https://pokeapi.co/api/v2/pokemon/3/",
      name: "venusaur",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png",
    },
    {
      __typename: "PokemonItem",
      id: 4,
      url: "https://pokeapi.co/api/v2/pokemon/4/",
      name: "charmander",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png",
    },
    {
      __typename: "PokemonItem",
      id: 5,
      url: "https://pokeapi.co/api/v2/pokemon/5/",
      name: "charmeleon",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png",
    },
  ],
};

describe("Pokemons", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });
  test("renders the component", () => {
    const element = render(
      <Pokemons pokemons={pokemons} fetchMore={jest.fn()} />
    );

    expect(element).toMatchSnapshot();
  });

  test("renders the pokemons", () => {
    const { getByText } = render(
      <Pokemons pokemons={pokemons} fetchMore={jest.fn()} />
    );

    expect(getByText("bulbasaur")).toMatchSnapshot();
    expect(getByText("ivysaur")).toMatchSnapshot();
    expect(getByText("venusaur")).toMatchSnapshot();
    expect(getByText("charmander")).toMatchSnapshot();
    expect(getByText("charmeleon")).toMatchSnapshot();
  });

  test("runs fetchMore function", () => {
    const fetchMore = jest.fn();
    const { getByText } = render(
      <Pokemons pokemons={pokemons} fetchMore={fetchMore} />
    );

    const button = getByText("Load more");

    fireEvent.click(button);

    expect(fetchMore).toHaveBeenCalledTimes(1);
    expect(fetchMore).toHaveBeenCalledWith({ variables: { offset: 5 } });
  });
});
