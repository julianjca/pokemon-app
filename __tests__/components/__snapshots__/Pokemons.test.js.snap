// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemons renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 1000px;
  margin: 0 auto;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),0 4px 6px -2px rgba(0, 0, 0, 0.05);
  border-radius: 1.5rem;
  position: relative;
  overflow: hidden;
  cursor: pointer;
  -webkit-transition: 0.1 all ease-in-out;
  transition: 0.1 all ease-in-out;
  background: #ffffff;
  color: #193569;
  margin: 0 10px 10px 0;
  height: 160px;
  width: 160px;
}

.emotion-2>span {
  font-size: 0.9rem;
  margin-top: 2px;
}

@media (min-width: 720px) {
  .emotion-2 {
    height: 180px;
    width: 180px;
    margin: 0 15px 15px 0;
  }
}

.emotion-2:hover {
  -webkit-transform: translateY(-3px);
  -moz-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
  -webkit-transition: 0.1s all ease-in-out;
  transition: 0.1s all ease-in-out;
}

.emotion-4 {
  font-family: Inter,sans-serif;
  font-weight: bold;
  text-transform: capitalize;
}

.emotion-6 {
  position: absolute;
  opacity: 0.08;
  bottom: -58px;
  right: -60px;
  height: 200px;
}

.emotion-32 {
  background-color: #193569;
  border-radius: 0.5em;
  border: none;
  padding: 1em 1.2em;
  font-size: 1rem;
  font-weight: bold;
  color: #FDC903;
  cursor: pointer;
  -webkit-transition: 0.15s all ease-in-out;
  transition: 0.15s all ease-in-out;
  outline: none;
}

.emotion-32:hover {
  background-color: #152c57;
  -webkit-transition: 0.15s all ease-in-out;
  transition: 0.15s all ease-in-out;
}

<body>
    <div>
      <div
        style="padding: 4rem 0px;"
      >
        <div
          class="emotion-0 emotion-1"
        >
          <a
            class="emotion-2 emotion-3"
          >
            <img
              alt="bulbasaur"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
            />
            <div
              class="emotion-4 emotion-5"
            >
              bulbasaur
            </div>
            <span>
              You have: 
              0
            </span>
            <img
              alt="Pokeball"
              class="emotion-6 emotion-7"
              src="/pokeball.svg"
            />
          </a>
          <a
            class="emotion-2 emotion-3"
          >
            <img
              alt="ivysaur"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
            />
            <div
              class="emotion-4 emotion-5"
            >
              ivysaur
            </div>
            <span>
              You have: 
              0
            </span>
            <img
              alt="Pokeball"
              class="emotion-6 emotion-7"
              src="/pokeball.svg"
            />
          </a>
          <a
            class="emotion-2 emotion-3"
          >
            <img
              alt="venusaur"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
            />
            <div
              class="emotion-4 emotion-5"
            >
              venusaur
            </div>
            <span>
              You have: 
              0
            </span>
            <img
              alt="Pokeball"
              class="emotion-6 emotion-7"
              src="/pokeball.svg"
            />
          </a>
          <a
            class="emotion-2 emotion-3"
          >
            <img
              alt="charmander"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
            />
            <div
              class="emotion-4 emotion-5"
            >
              charmander
            </div>
            <span>
              You have: 
              0
            </span>
            <img
              alt="Pokeball"
              class="emotion-6 emotion-7"
              src="/pokeball.svg"
            />
          </a>
          <a
            class="emotion-2 emotion-3"
          >
            <img
              alt="charmeleon"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png"
            />
            <div
              class="emotion-4 emotion-5"
            >
              charmeleon
            </div>
            <span>
              You have: 
              0
            </span>
            <img
              alt="Pokeball"
              class="emotion-6 emotion-7"
              src="/pokeball.svg"
            />
          </a>
        </div>
        <div
          style="width: 100%; text-align: center; margin-top: 2rem;"
        >
          <button
            class="emotion-32 emotion-33"
          >
            Load more
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 1000px;
  margin: 0 auto;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),0 4px 6px -2px rgba(0, 0, 0, 0.05);
  border-radius: 1.5rem;
  position: relative;
  overflow: hidden;
  cursor: pointer;
  -webkit-transition: 0.1 all ease-in-out;
  transition: 0.1 all ease-in-out;
  background: #ffffff;
  color: #193569;
  margin: 0 10px 10px 0;
  height: 160px;
  width: 160px;
}

.emotion-2>span {
  font-size: 0.9rem;
  margin-top: 2px;
}

@media (min-width: 720px) {
  .emotion-2 {
    height: 180px;
    width: 180px;
    margin: 0 15px 15px 0;
  }
}

.emotion-2:hover {
  -webkit-transform: translateY(-3px);
  -moz-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
  -webkit-transition: 0.1s all ease-in-out;
  transition: 0.1s all ease-in-out;
}

.emotion-4 {
  font-family: Inter,sans-serif;
  font-weight: bold;
  text-transform: capitalize;
}

.emotion-6 {
  position: absolute;
  opacity: 0.08;
  bottom: -58px;
  right: -60px;
  height: 200px;
}

.emotion-32 {
  background-color: #193569;
  border-radius: 0.5em;
  border: none;
  padding: 1em 1.2em;
  font-size: 1rem;
  font-weight: bold;
  color: #FDC903;
  cursor: pointer;
  -webkit-transition: 0.15s all ease-in-out;
  transition: 0.15s all ease-in-out;
  outline: none;
}

.emotion-32:hover {
  background-color: #152c57;
  -webkit-transition: 0.15s all ease-in-out;
  transition: 0.15s all ease-in-out;
}

<div>
    <div
      style="padding: 4rem 0px;"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <a
          class="emotion-2 emotion-3"
        >
          <img
            alt="bulbasaur"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
          />
          <div
            class="emotion-4 emotion-5"
          >
            bulbasaur
          </div>
          <span>
            You have: 
            0
          </span>
          <img
            alt="Pokeball"
            class="emotion-6 emotion-7"
            src="/pokeball.svg"
          />
        </a>
        <a
          class="emotion-2 emotion-3"
        >
          <img
            alt="ivysaur"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
          />
          <div
            class="emotion-4 emotion-5"
          >
            ivysaur
          </div>
          <span>
            You have: 
            0
          </span>
          <img
            alt="Pokeball"
            class="emotion-6 emotion-7"
            src="/pokeball.svg"
          />
        </a>
        <a
          class="emotion-2 emotion-3"
        >
          <img
            alt="venusaur"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
          />
          <div
            class="emotion-4 emotion-5"
          >
            venusaur
          </div>
          <span>
            You have: 
            0
          </span>
          <img
            alt="Pokeball"
            class="emotion-6 emotion-7"
            src="/pokeball.svg"
          />
        </a>
        <a
          class="emotion-2 emotion-3"
        >
          <img
            alt="charmander"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
          />
          <div
            class="emotion-4 emotion-5"
          >
            charmander
          </div>
          <span>
            You have: 
            0
          </span>
          <img
            alt="Pokeball"
            class="emotion-6 emotion-7"
            src="/pokeball.svg"
          />
        </a>
        <a
          class="emotion-2 emotion-3"
        >
          <img
            alt="charmeleon"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png"
          />
          <div
            class="emotion-4 emotion-5"
          >
            charmeleon
          </div>
          <span>
            You have: 
            0
          </span>
          <img
            alt="Pokeball"
            class="emotion-6 emotion-7"
            src="/pokeball.svg"
          />
        </a>
      </div>
      <div
        style="width: 100%; text-align: center; margin-top: 2rem;"
      >
        <button
          class="emotion-32 emotion-33"
        >
          Load more
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pokemons renders the pokemons 1`] = `
.emotion-0 {
  font-family: Inter,sans-serif;
  font-weight: bold;
  text-transform: capitalize;
}

<div
  class="emotion-0 emotion-1"
>
  bulbasaur
</div>
`;

exports[`Pokemons renders the pokemons 2`] = `
.emotion-0 {
  font-family: Inter,sans-serif;
  font-weight: bold;
  text-transform: capitalize;
}

<div
  class="emotion-0 emotion-1"
>
  ivysaur
</div>
`;

exports[`Pokemons renders the pokemons 3`] = `
.emotion-0 {
  font-family: Inter,sans-serif;
  font-weight: bold;
  text-transform: capitalize;
}

<div
  class="emotion-0 emotion-1"
>
  venusaur
</div>
`;

exports[`Pokemons renders the pokemons 4`] = `
.emotion-0 {
  font-family: Inter,sans-serif;
  font-weight: bold;
  text-transform: capitalize;
}

<div
  class="emotion-0 emotion-1"
>
  charmander
</div>
`;

exports[`Pokemons renders the pokemons 5`] = `
.emotion-0 {
  font-family: Inter,sans-serif;
  font-weight: bold;
  text-transform: capitalize;
}

<div
  class="emotion-0 emotion-1"
>
  charmeleon
</div>
`;
